src/Travers.php

```diff
@@ -65,6 +89,39 @@ public function find($key)
         return $this->parseParam($this->parseKeys($key), $this->tree);
     }
 
+    /**
+     * @param       $key
+     * @param array $tree
+     *
+     * @return array|mixed
+     *
+     * @throws BranchNotFoundException
+     */
+    protected function removeParam($key, array $tree)
+    {
+        $tmp       = &$tree;
+        $keysArray = $this->parseKeys($key);
+
+        while (count($keysArray) > 1) {
+            $key = array_shift($keysArray);
+
+            if (!is_array($tmp) || !array_key_exists($key, $tmp)) {
+                return $this->shouldFail() ? $this->handleFailure($key) : $tmp;
+            }
+
+            $tmp = &$tmp[$key];
+        }
+
+        $key = array_shift($keysArray);
+
+        if (!is_string($key) || !is_array($tmp) || !array_key_exists($key, $tmp)) {
+            return $this->handleFailure($key);
```

Comment:
if it is not configured to throw an exception, this will return null after a failed delete - it should return the tree.